ke/compiler.py

```diff
@@ -89,6 +89,47 @@
     builtin_macros["#" + short] = builtin_macros["#" + long]
 
 
+def compile_separate(separate, expr):
+    print('\n\n\n\n#########')
+
+    if separate is None:
+        raise CompileError("Must specify a valid separator: ',', ':', '|'")
+    print(f"incoming separator of: '{separate}' - '{expr}'")
+    (a, b, is_greedy, sub) = expr
+    if sub is None:
+        return CompileError("Must specify a token")
+
+    if b is not None and a > b:
+        return CompileError("Min can not exceed Max.")
+    
+    # we are ok with max of 0, we just send an empty response.
+    if b == 0:
+        return asm.Literal("") 
```

Comment:
we can use EMPTY in this file (it's the same, but communicates better)

Reply:
Resolved another similar occasion